lib/spack/spack/binary_distribution.py

```diff
@@ -0,0 +1,209 @@
+# "Benedikt Hegner (CERN)"
+
+import os
+import platform
+import yaml
+
+import llnl.util.tty as tty
+from llnl.util.filesystem import mkdirp
+
+from architecture import get_full_system_from_platform
+from spack.util.executable import which
+import spack.cmd
+import spack
+from spack.stage import Stage
+import spack.fetch_strategy as fs
+import spack.relocate
+
+
+def prepare():
+    """
+    Install patchelf as pre-requisite to the
+    required relocation of binary packages
+    """
+    if platform.system() == 'Darwin':
+        return
+    dir = os.getcwd()
+    patchelf_spec = spack.cmd.parse_specs("patchelf", concretize=True)[0]
+    if not spack.install_layout.check_installed(patchelf_spec):
```

Comment:
`spack.installed_db` has a `query()` method.  You can just run this:

```python
installed_specs = spack.installed_db.query('patchelf')
```

You probably really want to run:
```python
installed_specs = spack.installed_db.query('patchelf os=frontend')
```

to ensure you get a patchelf that matches the default frontend OS for the system (e.g. SuSE and not CNL on Cray).